client/support/support-user/localforage-bypass.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * External dependencies
+ */
+import debugModule from 'debug';
+
+const debug = debugModule( 'calypso:support-user' );
+
+// This module defines a custom localForage driver which bypasses all persistent
+// storage. Any calls to read/write data using localForage instead access a temporary
+// in-memory store which is lost on page reload. This driver is used to sandbox
+// a user's data while support-user is active, ensuring it does not contaminate the
+// original user/vice versa.
+// Copied with modifications from localForage source: localforage/test/dummyStorageDriver.js
+
+var dummyStorage = {};
+
+// Config the localStorage backend, using options set in the config.
+function _initStorage( options ) {
+	const self = this;
+
+	let dbInfo = {};
+	if ( options ) {
+		for ( let i in options ) {
+			dbInfo[ i ] = options[ i ];
+		}
+	}
+
+	dummyStorage[ dbInfo.name ] = dbInfo.db = {};
+
+	self._dbInfo = dbInfo;
+	return Promise.resolve();
+}
+
+function clear( callback ) {
+	debug( 'localForage bypass', 'clear' );
+
+	const self = this;
+	const promise = new Promise( function( resolve, reject ) {
+		self.ready().then( function() {
+			var db = self._dbInfo.db;
+
+			for ( let key in db ) {
+				if ( db.hasOwnProperty( key ) ) {
+					delete db[ key ];
+				}
+			}
+
+			resolve();
+		} ).catch( reject );
+	} );
+
+	executeCallback( promise, callback );
+	return promise;
+}
+
+function getItem( key, callback ) {
+	debug( 'localForage bypass', 'getItem', key );
+	const self = this;
+
+	// Cast the key to a string, as that's all we can set as a key.
+	if ( typeof key !== 'string' ) {
+		key = String( key );
+	}
+
+	const promise = new Promise( function( resolve, reject ) {
+		self.ready().then( function() {
+			try {
+				const db = self._dbInfo.db;
+				const result = db[ key ];
+
+				resolve( result );
+			} catch ( e ) {
+				reject( e );
+			}
+		} ).catch( reject );
+	} );
+
+	executeCallback( promise, callback );
+	return promise;
+}
+
+function iterate( callback ) {
+	const self = this;
+
+	const promise = new Promise( function( resolve, reject ) {
+		self.ready().then( function() {
+			try {
+				const db = self._dbInfo.db;
+
+				for ( let key in db ) {
+					const result = db[ key ];
+
+					callback( result, key );
```

Comment:
Does this actually iterate on following calls? Unit tests are highly suggested. :)

This might be a fun place to play with `yield`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield